### PR TITLE
Automation of viralrecon service setup and minor template changes.

### PR DESCRIPTION
**Pull request by jaimeozaez**
- Automation of viralrecon service setup. Modification of `buisciii_tools/bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon` file. This task is still under development.
- Increased reserved memory for task in IRMA nextflow srun (from 35000M to 350000) in order to prevent out-of-memory performance issues.
- Correction of 'DATE_entrega' directory numbering in assembly results lablog.
- Minor fixes in `buisciii_tools/bu_isciii/templates/IRMA/ANALYSIS/ANALYSIS01_FLU_IRMA/04-irma/` `lablog`.

### CODE REVIEW

File: bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon [hunk 1]
```diff
@@ -1,6 +1,97 @@
...
+    echo "Checking if latest version of Pangolin image is already downloaded..."
+    if [ -e "/data/bi/scratch_tmp/bi/singularity-images/$latest_version_pangolin" ]; then
+        echo "File $latest_version_pangolin already downloaded."
+    else
+        echo "Downloading $latest_version_pangolin file..."
+        wget -P "/data/bi/scratch_tmp/bi/singularity-images/" "https://depot.galaxyproject.org/singularity/$latest_version_pangolin"
+        echo "$latest_version_pangolin file succesfully downloaded."
+    fi
+
+    # Checking if latest_version_nextclade downloaded
```
Comment (saramonzon) [inline bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon:50]: RSV also has nextclade annotation, take this into account

File: bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon [hunk 2]
```diff
@@ -1,6 +1,97 @@
...
+    # Update viralrecon_sars_params.yml file
+    sed -i "s|2023-03-16T12:00:00Z|$nextclade_tag|" "../DOC/viralrecon_sars_params.yml"
+    echo "DATADIR in file ../DOC/viralrecon_sars_params.yml UPDATED."
+
+    # Setting work variables
+    ANALYSIS_TYPE=AMPLICONS
+    CONFIG_FILE="../../DOC/viralrecon_sars.config"
+    PARAMS_FILE="../../DOC/viralrecon_sars_params.yml"
+
+# Preparing the enviroment for non-SARS-CoV-2 viruses
```
Comment (saramonzon) [inline bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon:87]: I would add some checks for metagenomics, for examble the conf to export fastq files without host, and others I may not be remembering

File: bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon [hunk 3]
```diff
@@ -1,6 +1,97 @@
...
+        read -p "Please specify the reference: " reference
+        read -p "Are host [$host] and reference [$reference] correct? (y/n): " answer
+    done
+    python3 samples_ref_viralrecon.py $reference $host
+    echo "File samples_ref.txt READY."
+fi
+
+# Preparing the environment for SARS-CoV-2 analysis
+test_ref=$(cut -f2 samples_ref.txt | head -n1)
+if [ "$test_ref" == "NC_045512.2" ]; then
```
Comment (saramonzon) [inline bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon:22]: maybe i would ask here if metagenomics or amplicon before checking the reference? If amplicon for a known reference you can do that, but also make a base conf for the case it is an amplicon analysis with new amplicon primer beds.

File: bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon [hunk 4]
```diff
@@ -1,6 +1,97 @@
...
+if [ "$samples_ref_prepared" == "y" ]; then 
+    echo "File samples_ref.txt READY."
+else
+    echo "File samples_ref NOT prepared."
+    while [ -z "$host" ] || [ -z "$reference" ] || [ "$answer" != "y" ]; do
+        read -p "Please specify the host: " host
+        read -p "Please specify the reference: " reference
+        read -p "Are host [$host] and reference [$reference] correct? (y/n): " answer
+    done
+    python3 samples_ref_viralrecon.py $reference $host
```
[the remark this answers is not in the document]
Comment (jaimeozaez) [reply bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon:16]: Yes. I was just wondering where to place it (maybe in the same folder?)

File: bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon (hunk 1 above)
Comment (jaimeozaez) [reply bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon:50]: Thanks! I will include it in the non-SARS-CoV-2 section.

File: bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon (hunk 2 above)
Comment (jaimeozaez) [reply bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon:87]: Mmmm maybe we should talk about this before continuing.

File: bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon (hunk 3 above)
Comment (jaimeozaez) [reply bu_isciii/templates/viralrecon/ANALYSIS/lablog_viralrecon:22]: Got it! But we'll also need to talk about this.

File: CHANGELOG.md [hunk 5]
```diff
@@ -74,7 +74,7 @@ Code contributions to the hotfix:
 Code contributions to the hotfix:
 
 - [Pablo Mata](https://github.com/Shettland)
-- [Jaime Ozaez](https://github.com/jaimeozaez)
+- [Jaime Ozáez](https://github.com/jaimeozaez)
```
[the remark this answers is not in the document]
Comment (jaimeozaez) [reply CHANGELOG.md:85]: Done! Thanks

**Pull request closed without merging**